### ISSUE TITLE
Module and edgeHub scheduled for restart about once an hour by agent

### ISSUE THREAD

**Issue by kimhanse** [Azure/iotedge#1175]
While my module is running it gets restarted randomly every hour for no apparent reason.

## Expected Behavior

I would not expect my module to not get restarted.

## Current Behavior

My module gets restarted.

## Steps to Reproduce

1. Install IoT Edge on Rasperry Pi
2. Deploy module

### Device (Host) Operating System

Raspbian Stretch

### Architecture

arm32

### Container Operating System

Linux containers

#### iotedged

iotedge 1.0.6.1 (3fa6cbef8b7fc3c55a49a622735eb1021b8a5963)

#### Edge Agent

2019-04-30 11:49:20.370 +00:00 [INF] - Version - 1.0.6.19913336 (8288bc9bd6f6e15295fea506cd3f99d7f6347a6a)

#### Edge Hub

2019-05-06 08:35:11.765 +00:00 [INF] - Version - 1.0.6.19913336 (8288bc9bd6f6e15295fea506cd3f99d7f6347a6a)

#### Docker

Server:
 Engine:
  Version:          3.0.4

## Logs

```
pi@edge5:~ $ sudo docker logs --tail 100000 edgeAgent | grep 'scheduled to restart after' | tail -10
2019-05-06 06:25:36.928 +00:00 [INF] - Module 'edgeHub' scheduled to restart after 10s (02s left).
2019-05-06 06:25:36.929 +00:00 [INF] - Module 'ModbusMultiModule' scheduled to restart after 10s (02s left).
2019-05-06 07:09:16.048 +00:00 [INF] - Module 'edgeHub' scheduled to restart after 10s (08s left).
2019-05-06 07:09:16.048 +00:00 [INF] - Module 'ModbusMultiModule' scheduled to restart after 10s (08s left).
2019-05-06 07:09:21.302 +00:00 [INF] - Module 'edgeHub' scheduled to restart after 10s (03s left).
2019-05-06 07:09:21.302 +00:00 [INF] - Module 'ModbusMultiModule' scheduled to restart after 10s (03s left).
2019-05-06 08:34:37.320 +00:00 [INF] - Module 'edgeHub' scheduled to restart after 10s (07s left).
2019-05-06 08:34:37.321 +00:00 [INF] - Module 'ModbusMultiModule' scheduled to restart after 10s (07s left).
2019-05-06 08:34:42.573 +00:00 [INF] - Module 'edgeHub' scheduled to restart after 10s (02s left).
2019-05-06 08:34:42.574 +00:00 [INF] - Module 'ModbusMultiModule' scheduled to restart after 10s (02s left).
pi@edge5:~ $ 
```

## Additional Information

Just request the log files or any other information you will need to debug this and I will happily provide it.

**Comment by myagley** [comment]
Is it possible that your module is exiting? It will be helpful to look at the logs for the Edge Hub and your module.

The Edge Agent will only start a stopped/failed container. It doesn't proactively restart containers.

**Comment by kimhanse** [comment]
It might by our module that crashes at 05:42:00 in the AMQP library, but I didn't expect that to bring down the edgeHub at the same time.

Here are some log files around one crash that happened at 05:42:00-05:42:10:
* [agent.log](https://github.com/Azure/iotedge/files/3149506/agent.log)
* [hub.log](https://github.com/Azure/iotedge/files/3149504/hub.log)
* [module.log](https://github.com/Azure/iotedge/files/3149505/module.log)

**Comment by ancaantochi** [comment]
Hi @kimhanse,

Is edgeHub failing with the same exception or are there different exceptions in the logs?
Please update edgeHub and edgeAgent to the latest release 1.0.7 If the issue still reproduces, can you please turn on debug logs and send full logs of the modules? 

Here are instructions to turn on debug logs: https://docs.microsoft.com/en-us/azure/iot-edge/troubleshoot#view-the-messages-going-through-the-iot-edge-hub

**Comment by kimhanse** [comment]
Hi @ancaantochi ,

I don't think I get any data  from edge hub in the logs when it fails:
```
2019-05-06 05:38:37.342 +00:00 [INF] - Reauthenticating connected clients
[2019-05-06 05:42:13 +00:00]: Starting Edge Hub
```
The crashes happened at 05:42:00-05:42:10. I don't think that the exceptions in the logs are directly related to the crashes.

I am using the standard setup of `mcr.microsoft.com/azureiotedge-hub:1.0` and `mcr.microsoft.com/azureiotedge-agent:1.0`, do I need to do something to upgrade to the latest release?

**Comment by ancaantochi** [comment]
Hi, 

Do you have debug logs turned on and no exception in edgeHub and your module before the restart?

Please set in the deployment 1.0.7 version: `mcr.microsoft.com/azureiotedge-hub:1.0.7` and `mcr.microsoft.com/azureiotedge-agent:1.0.7`

**Comment by kimhanse** [comment]
I am doing a full upgrade of my system to 1.0.7 including upgrading Raspian etc. I will return in a few days when I am done with the results

**Comment by kimhanse** [comment]
Log files for the minute where it fails, this time with debug logging enabled:
* [edgeHub-2019-05-20T09:03.log](https://github.com/Azure/iotedge/files/3197187/edgeHub-2019-05-20T09.03.log)
* [edgeAgent-2019-05-20T09:03.log](https://github.com/Azure/iotedge/files/3197188/edgeAgent-2019-05-20T09.03.log)
* [ModbusMultiModule-2019-05-20T09:03.log](https://github.com/Azure/iotedge/files/3197189/ModbusMultiModule-2019-05-20T09.03.log)

**Comment by lt72** [comment]
Hello, we fixed a few issues is this area with 1.0.8 and 1.0.9. If you can try any of those versions, please re-open the issue if it still repros.
